src/utils/api/account/lsk.js

```diff
@@ -79,26 +80,37 @@ export const getAccount = async ({
       params: normParams,
     });
 
-    return response.data[0];
-  } catch (e) {
-    // eslint-disable-next-line no-console
-    console.log('Lisk account not found.');
-    let publicKey = params.publicKey;
-    if (!publicKey && params.passphrase) {
-      publicKey = extractPublicKey(params.passphrase);
+    if (response.data[0]) {
+      const account = { ...response.data[0] };
+      const isAccountUninitialized = !account.summary.publicKey;
+      if (isAccountUninitialized && (params.publicKey || params.passphrase)) {
+        const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
+        account.summary.publicKey = publicKey;
+      }
+      return account;
     }
+  } catch (e) {
+    if (e.code === HTTP_CODES.NOT_FOUND) {
+      // eslint-disable-next-line no-console
+      console.log('Lisk account not found.');
 
-    const account = {
-      summary: {
-        publicKey,
-        balance: 0,
-        address: normParams.address,
-        token: tokenMap.LSK.key,
-      },
-    };
-
-    return account;
+      if (params.publicKey || params.passphrase) {
+        const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
+        const account = {
+          summary: {
+            publicKey,
+            balance: 0,
+            address: normParams.address,
+            token: tokenMap.LSK.key,
+          },
+        };
+        return account;
+      }
+    } else {
+      throw Error();
+    }
   }
+  throw Error('Error retrieving account');
```

Comment:
In what condition the code reaches here?

Reply:
Ideally, never. But to keep the code branches clean, I added this error. There is also the possibility of a network failure and then the code execution will end up here.